reference/client-access/python-sdk.md

```diff
@@ -0,0 +1,287 @@
+---
+description: Technical reference for the libSQL Python client library used to access Turso databases, including sample code.
+keywords:
+  - turso
+  - libsql
+  - python
+  - client
+  - sdk
+  - library
+  - example
+---
+
+# Python SDK
+
+The Python SDK comes with APIs for both synchronous and asynchronous code.
+
+## Installation
+
+Add the [libsql_client module] to your project using `pip`:
+
+```bash
+$ pip install libsql_client
+```
+
+## Initialization
+
+Import the libsql_client module:
+
+```py
+import libsql_client
+```
+
+If you want to use the synchronous API, use the `create_client_sync` function:
+
+```py
+client = libsql_client.create_client_sync(
+    url="libsql://your-database.turso.io"
+    auth_token="your-auth-token"
+)
+```
+
+If you want to use the asynchronous API, use the `create_client` function along
+with `asyncio` and async/await syntax:
+
+```py
+client = libsql_client.create_client_sync(
+    url="libsql://your-database.turso.io"
+    auth_token="your-auth-token"
+)
+```
+
+The `auth_token` parameter is only required when using a remote database
+instance managed by Turso.
+
+In either case, the returned Client object exposes an identical API. The
+synchronous client is a thin wrapper around the asynchronous client.
+
+The examples on this page use the synchronous API. The equivalent async code can
+be easily derived from that by applying `async` and `await`.  For example, a
+synchronous call to the client's `execute()` method takes this form:
```

Comment:
Maybe we should nudge users towards using the async API by default?

Also, moving from the async API to sync API is easier, because you can just drop `await` and `async`. In the other direction, from sync to async API, you have to guess: for example, with the async client, you should use `async with` with the client, but `with` (not `async with`) with the transaction.

Reply:
I'm firmly convinced that most python users are going to prefer the sync client and would be potentially confused by the extra async/await keywords.  Similarly, someone who does prefer the async client would likely have a better idea what to do here to make it work correctly. So I'm choosing to first help the less experienced folks, with the understanding the that more experienced folks will be able to help themselves.